components/canvas/gl_context.rs

```diff
@@ -168,44 +193,101 @@ impl GLContextWrapper {
             GLContextWrapper::OSMesa(ref ctx) => {
                 WebGLImpl::apply(ctx, state, cmd, backtrace);
             },
+            #[cfg(target_os = "macos")]
+            GLContextWrapper::NativeWithIOSurface(ref ctx) => {
+                WebGLImpl::apply(ctx, state, cmd, backtrace);
+            },
         }
     }
 
     pub fn gl(&self) -> &dyn gl::Gl {
         match *self {
             GLContextWrapper::Native(ref ctx) => ctx.gl(),
             GLContextWrapper::OSMesa(ref ctx) => ctx.gl(),
+            #[cfg(target_os = "macos")]
+            GLContextWrapper::NativeWithIOSurface(ref ctx) => ctx.gl(),
         }
     }
 
-    pub fn get_info(&self) -> (Size2D<i32>, u32, GLLimits) {
+    pub fn get_info(&self) -> (Size2D<i32>, u32, Option<u32>, GLLimits) {
         match *self {
             GLContextWrapper::Native(ref ctx) => {
-                let (real_size, texture_id) = {
+                let (real_size, texture_id, io_surface_id) = {
                     let draw_buffer = ctx.borrow_draw_buffer().unwrap();
                     (
                         draw_buffer.size(),
                         draw_buffer.get_bound_texture_id().unwrap(),
```

Comment:
If we make get_bound_texture_id return a TextureBacking object, we don't need to create a separate `io_surface_id` value.